Fix compiler warning in WTF::moveToVectorOf()

cdumez opened the pull request:
#### 9019b5fba9140ef43a39d0855594b638fbf8f4f9
<pre>
Fix compiler warning in WTF::moveToVectorOf()
<a href="https://bugs.webkit.org/show_bug.cgi?id=261534">https://bugs.webkit.org/show_bug.cgi?id=261534</a>

Reviewed by Darin Adler.

Fix compiler warning in WTF::moveToVectorOf():
```
/usr/local/include/wtf/Vector.h:1935:16: warning: Forwarding reference passed
to std::move(), which may unexpectedly cause lvalues to be moved; use
std::forward() instead [bugprone-move-forwarding-reference]
```

* Source/WTF/wtf/Vector.h:
(WTF::moveToVectorOf):

Canonical link: <a href="https://commits.webkit.org/267994@main">https://commits.webkit.org/267994@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9f31ec0bb4c6e10c8bc4e38f971ebbc60fba4e7a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18282 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18618 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19197 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20123 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17112 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21916 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18771 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19048 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18505 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18713 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15924 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21006 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15936 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16677 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/23175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/15842 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16955 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16848 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/21065 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/17568 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17412 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/14770 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/21640 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16508 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5301 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4357 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20871 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/22874 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17255 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5147 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/Vector.h:
```diff
@@ -1932,7 +1932,7 @@ template<typename DestinationItemType, typename Collection>
 {
     return WTF::map(collection, [] (auto&& v) -> DestinationItemType {
-        return WTFMove(v);
+        return std::forward<DestinationItemType>(v);
```
The use of `v` for this local variable name, which is not a vector, is unnecessarily confusing. I suggest renaming to `item`.

cdumez committed:
Fix compiler warning in WTF::moveToVectorOf()
https://bugs.webkit.org/show_bug.cgi?id=261534

Reviewed by Darin Adler.

Fix compiler warning in WTF::moveToVectorOf():
```
/usr/local/include/wtf/Vector.h:1935:16: warning: Forwarding reference passed
to std::move(), which may unexpectedly cause lvalues to be moved; use
std::forward() instead [bugprone-move-forwarding-reference]
```

* Source/WTF/wtf/Vector.h:
(WTF::moveToVectorOf):

Canonical link: https://commits.webkit.org/267994@main
